core/src/main/java/tech/tablesaw/api/Table.java

```diff
@@ -864,10 +864,24 @@ public Table countBy(CategoricalColumn<?> groupingColumn) {
         return groupingColumn.countByCategory();
     }
 
+    /**
+     * Join on the given {@code columnName}  
+     * @param columnName			The column name to join on
+     */
     public DataFrameJoiner join(String columnName) {
         return new DataFrameJoiner(this, columnName);
     }
 
+    /**
+     * Join on the given {@code columnName} with {@code joinTableInitialId} as the initial
+     * index used when renaming duplicate columns. 
+     * @param columnName			The column name to join on
+     * @param joinTableInitialId	The initial index used when naming duplicate columns, e.g. "T3.Age"
+     */
+    public DataFrameJoiner join(String columnName, int joinTableInitialId) {
```

Comment:
I don't like the idea of making `joinTableInitialId` part of the public interface as it seems like it's exposing too much of the internals. Can we come up with another solution like checking if the column exists after incrementing the ID and then incrementing again if necessary?